### DISCUSSION TITLE
Sensing matrix is not full rank

### DISCUSSION THREAD

**Discussion by LiuHaoXuan0403** [ttadano/alamode#105] [Q&A]
Hi Dr. Tadano,
I try to use ALAMODE to fit the harmonic FCs using alm. However, the following errors are output:
 
![Uploading image.png…]()

I have use all disp.POSCARs recommended by alm to calculate the forces. How can I increase the number of training datasets? Could you give me a clue for solving this problem?

**Comment by ttadano** [comment]
One of the easiest ways is to use the same displacement patterns with a different displacement magnitude (specified by `--mag`).

**Comment by LiuHaoXuan0403** [comment]
Dear Dr. Tadano
Thank you very much for the suggestion.
I use the same displacement patterns with a different displacement magnitude (--mag =0.01,0.02 and 0.03). The problem has been solved. However, the fitting error is beyond 20% for harmonic FCs. How can I further increase the number of training datasets to reduce the error？

**Comment by gangliu-github** [comment]
What material does you deal with? I encounter the same problem now.
